hack/arktos_cherrypick.sh

```diff
@@ -0,0 +1,210 @@
+    git diff --cached --name-only --diff-filter=A | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS|arktos_copyright" >> $LOGDIR/added_files || true
+    #grep -F -x -v -f $REPODIRNAME/hack/arktos_copyright_copied_k8s_files $LOGDIR/added_files_git > $LOGDIR/added_files_less_copied
+    #grep -F -x -v -f $REPODIRNAME/hack/arktos_copyright_copied_modified_k8s_files $LOGDIR/added_files_less_copied > $LOGDIR/added_files
+    popd
+}
+
+get_modified_files_list() {
+    pushd $REPODIRNAME
+    local MERGED_COMMIT=$( git log --show-signature --oneline | grep "gpg: Signature made" | head -n 1 | cut -c1-7 )
+    local HEAD_COMMIT=`git rev-list HEAD | head -n 1`
+    echo "MERGED_COMMIT: $MERGED_COMMIT, HEAD_COMMIT: $HEAD_COMMIT"
+    git diff --name-only --diff-filter=M $MERGED_COMMIT $HEAD_COMMIT | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "perf-tests\/clusterloader2\/" | \
+        egrep -v "staging\/src\/k8s.io\/component-base\/metrics\/" | \
+        egrep -v "staging\/src\/k8s.io\/component-base\/version" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS" > $LOGDIR/changed_files
+    git diff --cached --name-only --diff-filter=M | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS|arktos_copyright" >> $LOGDIR/changed_files || true
+    #cat $REPODIRNAME/hack/arktos_copyright_copied_modified_k8s_files >> $LOGDIR/changed_files
+    popd
+}
+
+check_and_add_arktos_cherrypick() {
+    local REPOFILE=$1
+    set +e
+    cat $REPOFILE | grep "$ARKTOS_CHERRYPICK_MATCH" > /dev/null 2>&1
+    if [ $? -eq 0 ]
+    then
+        local tstamp=$($STAT_CMD $REPOFILE)
+        if [[ $REPOFILE = *.go ]] || [[ $REPOFILE = *.proto ]]
+        then
+            $SED_CMD -i "/$ARKTOS_CHERRYPICK_MATCH/a $ARKTOS_CHERRYPICK_LINE_GO" $REPOFILE
+        else
+            $SED_CMD -i "/$ARKTOS_CHERRYPICK_MATCH/a $ARKTOS_CHERRYPICK_LINE_OTHER" $REPOFILE
+        fi
+        $TOUCH_CMD "$tstamp" $REPOFILE
+    else    
+        echo "ERROR: File $REPOFILE does not have either K8s or Arktos copyright." >> $LOGFIL
+    fi
+    set -e
+}
+
+add_arktos_cherrypick() {
+    echo "Inspecting cherrypick files, writing logs to $LOGFILE"
+    local ADDED_FILELIST=$1
+    local CHANGED_FILELIST=$2
+    while IFS= read -r line
+    do
+        if [ ! -z $line ]
+        then
+            check_and_add_arktos_cherrypick $REPODIRNAME/$line
+        fi
+    done < $CHANGED_FILELIST
+    while IFS= read -r line
+    do
+        if [ ! -z $line ]
+        then
+            check_and_add_arktos_cherrypick $REPODIRNAME/$line
+        fi
+    done < $ADDED_FILELIST
+    echo "Done."
+}
+
+if [ -z $1 ]
+then
+    setup_repos
+fi
+
+get_added_files_list
+get_modified_files_list
+
+add_arktos_cherrypick $LOGDIR/added_files $LOGDIR/changed_files
```

Comment:
Does this end up adding cherrypick line to all added and modified files? Or does it apply only to the new files that were cherrypicked?

Reply:
This applied to both new added/updated files

Reply:
As per yesterday's discussion, I understand this script will add cherry-pick comment only to newly added files, and it won't modify existing files that were added by us. If that is the case, it looks good to me.